sdk/identity/azure-identity/src/main/java/com/azure/identity/IntelliJCredential.java

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.implementation.IdentityClient;
+import com.azure.identity.implementation.IdentityClientBuilder;
+import com.azure.identity.implementation.IdentityClientOptions;
+import com.azure.identity.implementation.MsalToken;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A credential provider that provides token credentials based on Azure CLI
+ * command.
+ */
+@Immutable
+public class IntelliJCredential implements TokenCredential {
```

Comment:
I am not sure why this is required. Is this specific to an IDE? It would be good to add some description of why this change is needed.